tests/nexus/nexus_workflow_test.go

```diff
@@ -1380,6 +1383,101 @@ func (s *NexusWorkflowTestSuite) TestNexusOperationAsyncCompletionAfterReset() {
+	handlerWorkflowTaskQueue := testcore.RandomizeStr("handler_" + s.T().Name())
+	endpointName := testcore.RandomizedNexusEndpoint(s.T().Name())
+	handlerWorkflowID := testcore.RandomizeStr(s.T().Name())
+
+	_, err := s.SdkClient().OperatorService().CreateNexusEndpoint(ctx, &operatorservice.CreateNexusEndpointRequest{
+		Spec: &nexuspb.EndpointSpec{
+			Name: endpointName,
+			Target: &nexuspb.EndpointTarget{
+				Variant: &nexuspb.EndpointTarget_Worker_{
+					Worker: &nexuspb.EndpointTarget_Worker{
+						Namespace: s.Namespace(),
+						TaskQueue: callerTaskQueue,
+					},
+				},
+			},
+		},
+	})
+	s.NoError(err)
+
+	w := worker.New(
+		s.SdkClient(),
+		callerTaskQueue,
+		worker.Options{},
+	)
+
+	svc := nexus.NewService("test")
+	wf := func(ctx workflow.Context, input nexus.NoValue) (nexus.NoValue, error) {
+		return nil, nil
+	}
+	op := temporalnexus.NewWorkflowRunOperation("op", wf, func(ctx context.Context, nv nexus.NoValue, opts nexus.StartOperationOptions) (client.StartWorkflowOptions, error) {
+		return client.StartWorkflowOptions{
+			ID:        handlerWorkflowID,
+			TaskQueue: handlerWorkflowTaskQueue,
+		}, nil
+	})
+	svc.Register(op)
+
+	callerWF := func(ctx workflow.Context, input nexus.NoValue) (nexus.NoValue, error) {
+		c := workflow.NewNexusClient(endpointName, svc.Name)
+		fut := c.ExecuteOperation(ctx, op, nil, workflow.NexusOperationOptions{})
+		return nil, fut.Get(ctx, nil)
+	}
+
+	w.RegisterNexusService(svc)
+	w.RegisterWorkflow(wf)
+	w.RegisterWorkflow(callerWF)
+	w.Start()
+	defer w.Stop()
+
+	run, err := s.SdkClient().ExecuteWorkflow(ctx, client.StartWorkflowOptions{
+		TaskQueue: callerTaskQueue,
+	}, callerWF, nil)
+	s.NoError(err)
+
+	pollRes, err := s.FrontendClient().PollWorkflowTaskQueue(ctx, &workflowservice.PollWorkflowTaskQueueRequest{
+		Namespace: s.Namespace(),
+		TaskQueue: &taskqueuepb.TaskQueue{
+			Name: handlerWorkflowTaskQueue,
+		},
+		Identity: "test",
+	})
+	s.NoError(err)
+	_, err = s.FrontendClient().RespondWorkflowTaskCompleted(ctx, &workflowservice.RespondWorkflowTaskCompletedRequest{
+		Namespace: s.Namespace(),
+		TaskToken: pollRes.TaskToken,
+		Identity:  "test",
+		Commands: []*commandpb.Command{
+			{
+				CommandType: enumspb.COMMAND_TYPE_COMPLETE_WORKFLOW_EXECUTION,
+				Attributes: &commandpb.Command_CompleteWorkflowExecutionCommandAttributes{
+					CompleteWorkflowExecutionCommandAttributes: &commandpb.CompleteWorkflowExecutionCommandAttributes{
+						Result: nil, // Return nil here to verify that the conversion to nexus content works as expected.
+					},
+				},
+			},
+		},
+	})
+	s.NoError(err)
+
+	s.NoError(run.Get(ctx, nil))
+	history := s.SdkClient().GetWorkflowHistory(ctx, run.GetID(), "", false, enumspb.HISTORY_EVENT_FILTER_TYPE_ALL_EVENT)
+	for history.HasNext() {
+		ev, err := history.Next()
+		s.NoError(err)
+		if attr := ev.GetNexusOperationCompletedEventAttributes(); attr != nil {
+			protorequire.ProtoEqual(s.T(), s.mustToPayload(nil), attr.GetResult())
```

Comment:
nit: `s.EqualExportedValues()` does the same without custom code.

Reply:
I'm not sure this works for protos. `protorequire` was developed specifically for proto comparison when we transitioned from gogo to google proto.